docs/user-guide/concepts/agents/conversation-management.md

```diff
@@ -87,8 +87,44 @@ Key features of the `SlidingWindowConversationManager`:
 - **Dangling Message Cleanup**: Removes incomplete message sequences to maintain valid conversation state.
 - **Overflow Trimming**: In the case of a context window overflow, it will trim the oldest messages from history until the request fits in the models context window.
 - **Configurable Tool Result Truncation**: Enable / disable truncation of tool results when the message exceeds context window limits. When `should_truncate_results=True` (default), large results are truncated with a placeholder message. When `False`, full results are preserved but more historical messages may be removed.
+- **Per-Turn Management**: Optionally apply context management proactively during the agent loop execution, not just at the end.
 
-#### SummarizingConversationManager
+**Per-Turn Management**:
+
+By default, the `SlidingWindowConversationManager` applies context management only after the agent loop completes. The `per_turn` parameter allows you to proactively manage context during execution, which is useful for long-running agent loops with many tool calls.
+
+=== "Python"
+
+    ```python
+    from strands import Agent
+    from strands.agent.conversation_manager import SlidingWindowConversationManager
+
+    # Apply management before every model call
+    conversation_manager = SlidingWindowConversationManager(
+        per_turn=True,  # Apply management before each model call
+    )
+
+    # Or apply management every N model calls
+    conversation_manager = SlidingWindowConversationManager(
+        per_turn=3,  # Apply management every 3 model calls
+    )
+
+    agent = Agent(
+        conversation_manager=conversation_manager
+    )
+    ```
+
+{{ ts_not_supported_code() }}
+
+The `per_turn` parameter accepts:
+
+- `False` (default): Only apply management after the agent loop completes
+- `True`: Apply management before every model call
+- An integer `N` (must be > 0): Apply management every N model calls
+
+This feature leverages the conversation manager's ability to act as a [HookProvider](#conversation-managers-as-hookproviders), registering for the `BeforeModelCallEvent` to proactively trim context during execution.
```

Comment:
Does the `conversation-managers-as-hookproviders` section resolve to anywhere? I couldn't find it.